languages/csharp/AppEncryption/AppEncryption/Persistence/AdoMetastorePersistenceImpl.cs

```diff
@@ -123,16 +121,16 @@ public bool Store(string keyId, DateTimeOffset created, JObject value)
 
                             int result = command.ExecuteNonQuery();
 
-                            // Database inserts with WHERE NOT EXISTS sub-clause will return 0 on a duplicate key insertion.
-                            // So, we can return false for such cases
                             return result == 1;
                         }
                     }
                 }
                 catch (DbException dbe)
                 {
                     Logger.LogError(dbe, "Metastore error during store");
-                    throw new AppEncryptionException("Metastore error:", dbe);
+
+                    // Even if the query failed due to a duplicate record return false
```

Comment:
may be worth expanding/clarifying this comment slightly. basically conveying that ado doesn't have a specific integrity violation error code/exception and we decided to always return false even for systemic issues for simplicity (system errors in select queries will blow up anyway)